make ProcessGroup.Options.timeout argument private in python

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56549 make ProcessGroupDefaultTimeout the same as python
* **#56531 make ProcessGroup.Options.timeout argument private in python**

per discussions in
https://github.com/pytorch/pytorch/pull/53663/files#r593409009, we need
to make sure our API not confusing user by passing in both timeout in
argument and timeout in processgroup.options. This PR tries to make the
`ProcessGroup.Options.timeout` be a private field, and only be used in
our test utils, for both `init_process_group` and `new_group`, we still
allow user pass `timeout` as a separate argument. Since
`ProcessGroupGloo.Options` only have a `timeout` config, both functions
will not allow passing in options for the GLOO backend.

This way we still preserve the only `timeout` API, and only allow user
to use `ProcessGroupNCCL.Options` when needed.

cc @pritamdamania87 @rohan-varma

Differential Revision: [D27893395](https://our.internmc.facebook.com/intern/diff/D27893395)

wanchaol committed:
make ProcessGroup.Options.timeout argument private in python

[ghstack-poisoned]

wanchaol committed:
Update on "make ProcessGroup.Options.timeout argument private in python"


per discussions in
https://github.com/pytorch/pytorch/pull/53663/files#r593409009, we need
to make sure our API not confusing user by passing in both timeout in
argument and timeout in processgroup.options. This PR tries to make the
`ProcessGroup.Options.timeout` be a private field, and only be used in
our test utils, for both `init_process_group` and `new_group`, we still
allow user pass `timeout` as a separate argument. Since
`ProcessGroupGloo.Options` only have a `timeout` config, both functions
will not allow passing in options for the GLOO backend.

This way we still preserve the only `timeout` API, and only allow user
to use `ProcessGroupNCCL.Options` when needed.

cc @pritamdamania87 @rohan-varma

[ghstack-poisoned]

wanchaol committed:
Update on "make ProcessGroup.Options.timeout argument private in python"


per discussions in
https://github.com/pytorch/pytorch/pull/53663/files#r593409009, we need
to make sure our API not confusing user by passing in both timeout in
argument and timeout in processgroup.options. This PR tries to make the
`ProcessGroup.Options.timeout` be a private field, and only be used in
our test utils, for both `init_process_group` and `new_group`, we still
allow user pass `timeout` as a separate argument. Since
`ProcessGroupGloo.Options` only have a `timeout` config, both functions
will not allow passing in options for the GLOO backend.

This way we still preserve the only `timeout` API, and only allow user
to use `ProcessGroupNCCL.Options` when needed.

cc @pritamdamania87 @rohan-varma

Differential Revision: [D27893395](https://our.internmc.facebook.com/intern/diff/D27893395)

[ghstack-poisoned]

wanchaol committed:
Update on "make ProcessGroup.Options.timeout argument private in python"


per discussions in
https://github.com/pytorch/pytorch/pull/53663/files#r593409009, we need
to make sure our API not confusing user by passing in both timeout in
argument and timeout in processgroup.options. This PR tries to make the
`ProcessGroup.Options.timeout` be a private field, and only be used in
our test utils, for both `init_process_group` and `new_group`, we still
allow user pass `timeout` as a separate argument. Since
`ProcessGroupGloo.Options` only have a `timeout` config, both functions
will not allow passing in options for the GLOO backend.

This way we still preserve the only `timeout` API, and only allow user
to use `ProcessGroupNCCL.Options` when needed.

cc @pritamdamania87 @rohan-varma

Differential Revision: [D27893395](https://our.internmc.facebook.com/intern/diff/D27893395)

[ghstack-poisoned]

rohan-varma commented on torch/csrc/distributed/c10d/init.cpp:
```diff
@@ -1256,31 +1233,18 @@ Example::
     >>> import torch.distributed as dist
-    >>> from datetime import timedelta
     >>>
-    >>> nccl_options = dist.ProcessGroupNCCL.Options(timeout=timedelta(seconds=300))
```
Nit: in the doc above (L1236), let's add that is_high_priority_stream is false by default?

rohan-varma commented on torch/csrc/distributed/c10d/init.cpp:
```diff
@@ -1126,30 +1125,8 @@ distributed: (:class:`~torch.distributed.ProcessGroupGloo.Options` and
-      .def(
-          py::init<std::chrono::milliseconds>(),
-          py::arg("timeout") = kProcessGroupDefaultTimeout)
+      processGroupGloo, "Options", processGroupOptions)
```
Since `c10d.ProcessGroupGloo.Options` is no longer supported should we prefix it with a _? `c10d.ProcessGroupGloo._Options`?

rohan-varma commented on torch/distributed/distributed_c10d.py:
```diff
@@ -463,9 +462,6 @@ def init_process_group(backend,
             backend, ``is_high_priority_stream`` can be specified so that
             process group can pick up high priority cuda streams.
 
-    .. note:: Note that if passing in pg_options and set the ``pg_options.timeout``,
```
Above for the doc about `pg_options`, should we mention that only `ProcessGroupNCCLOptions` should be used?

rohan-varma reviewed (approved):
LGTM, code is a lot cleaner now that we've removed timeout!

wanchaol committed:
Update on "make ProcessGroup.Options.timeout argument private in python"


per discussions in
https://github.com/pytorch/pytorch/pull/53663/files#r593409009, we need
to make sure our API not confusing user by passing in both timeout in
argument and timeout in processgroup.options. This PR tries to make the
`ProcessGroup.Options.timeout` be a private field, and only be used in
our test utils, for both `init_process_group` and `new_group`, we still
allow user pass `timeout` as a separate argument. Since
`ProcessGroupGloo.Options` only have a `timeout` config, both functions
will not allow passing in options for the GLOO backend.

This way we still preserve the only `timeout` API, and only allow user
to use `ProcessGroupNCCL.Options` when needed.

cc @pritamdamania87 @rohan-varma

Differential Revision: [D27893395](https://our.internmc.facebook.com/intern/diff/D27893395)

[ghstack-poisoned]

wanchaol committed:
Update on "make ProcessGroup.Options.timeout argument private in python"


per discussions in
https://github.com/pytorch/pytorch/pull/53663/files#r593409009, we need
to make sure our API not confusing user by passing in both timeout in
argument and timeout in processgroup.options. This PR tries to make the
`ProcessGroup.Options.timeout` be a private field, and only be used in
our test utils, for both `init_process_group` and `new_group`, we still
allow user pass `timeout` as a separate argument. Since
`ProcessGroupGloo.Options` only have a `timeout` config, both functions
will not allow passing in options for the GLOO backend.

This way we still preserve the only `timeout` API, and only allow user
to use `ProcessGroupNCCL.Options` when needed.

cc @pritamdamania87 @rohan-varma

Differential Revision: [D27893395](https://our.internmc.facebook.com/intern/diff/D27893395)

[ghstack-poisoned]
